File: hphp/hack/src/hhbc/hhas_adata.ml
(*
 * Copyright (c) 2017, Facebook, Inc.
 * All rights reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the "hack" directory of this source tree.
 *
 *)

type t = {
  adata_id: Hhbc_ast.adata_id;
  adata_value: Typed_value.t;
}

let make adata_id adata_value = { adata_id; adata_value }

let id adata = adata.adata_id

let value adata = adata.adata_value

File: hphp/hack/src/hhbc/hhas_adata.rs
// Copyright (c) Facebook, Inc. and its affiliates.
//
// This source code is licensed under the MIT license found in the
// LICENSE file in the "hack" directory of this source tree.
use runtime::TypedValue;

pub const VARRAY_PREFIX: &str = "y";
pub const DARRAY_PREFIX: &str = "Y";
pub const VEC_PREFIX: &str = "v";
pub const DICT_PREFIX: &str = "D";
pub const KEYSET_PREFIX: &str = "k";

#[derive(Debug)]
pub struct HhasAdata {
    pub id: String,
    pub value: TypedValue,
}
